corehq/apps/events/tests/test_views.py

```diff
@@ -0,0 +1,183 @@
+            cls.password,
+            None,
+            None,
+            is_admin=True,
+        )
+        cls.admin_webuser.save()
+
+        cls.non_admin_webuser = WebUser.create(
+            cls.domain,
+            cls.non_admin_web_username,
+            cls.password,
+            None,
+            None,
+            is_admin=False,
+        )
+        cls.non_admin_webuser.save()
+
+        cls.role_webuser = WebUser.create(
+            cls.domain,
+            cls.role_web_username,
+            cls.password,
+            None,
+            None,
+            is_admin=False,
+        )
+        role = cls.attendance_coordinator_role()
+        cls.role_webuser.set_role(cls.domain, role.get_qualified_id())
+        cls.role_webuser.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.admin_webuser.delete(None, None)
+        cls.non_admin_webuser.delete(None, None)
+        cls.role_webuser.delete(None, None)
+        cls.domain_obj.delete()
+
+        user_roles = UserRole.objects.filter(
+            name=UserRolePresets.ATTENDANCE_COORDINATOR, domain=cls.domain
+        )
+        if user_roles:
+            user_roles[0].delete()
+
+        super().tearDownClass()
+
+    @classmethod
+    def attendance_coordinator_role(cls):
+        return UserRole.create(
+            cls.domain,
+            UserRolePresets.ATTENDANCE_COORDINATOR,
+            permissions=HqPermissions(manage_attendance_tracking=True),
+        )
+
+    def log_user_in(self, user):
+        self.client.login(
+            username=user.username,
+            password=self.password,
+        )
+
+    @property
+    def endpoint(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+
+class TestEventsListView(BaseEventViewTestClass):
+
+    urlname = EventsView.urlname
+
+    @flag_enabled('ATTENDANCE_TRACKING')
+    def test_user_does_not_have_permission(self):
+        self.log_user_in(self.non_admin_webuser)
+
+        response = self.client.get(self.endpoint)
+        self.assertEqual(response.status_code, 404)
+
+    @flag_enabled('ATTENDANCE_TRACKING')
+    def test_user_is_domain_admin(self):
+        self.log_user_in(self.admin_webuser)
+
+        response = self.client.get(self.endpoint)
+        self.assertEqual(response.status_code, 200)
+
+    @flag_enabled('ATTENDANCE_TRACKING')
+    def test_non_admin_user_with_appropriate_role(self):
+        self.log_user_in(self.role_webuser)
+
+        response = self.client.get(self.endpoint)
+        self.assertEqual(response.status_code, 200)
+
+
+class TestEventsCreateView(BaseEventViewTestClass):
+
+    urlname = EventCreateView.urlname
+
+    @flag_enabled('ATTENDANCE_TRACKING')
+    @patch.object(Event, 'save')
+    def test_user_does_not_have_permission(self, event_save_method):
+        self.log_user_in(self.non_admin_webuser)
+
+        response = self.client.post(self.endpoint, self._event_data())
+        self.assertEqual(response.status_code, 404)
```

Comment:
Semi-nit: I'd advocate for using django status codes rather than magic numbers.
```python
from rest_framework import status

self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```